src/System.Private.ServiceModel/src/System/ServiceModel/Channels/SecurityChannelFactory.cs

```diff
@@ -0,0 +1,843 @@
+                if (!SecurityUtils.IsSecurityBindingSuitableForChannelBinding(SecurityProtocolFactory.SecurityBindingElement as TransportSecurityBindingElement))
+                    return;
+
+                if (InnerChannel != null)
+                {
+                    IChannelBindingProvider cbp = InnerChannel.GetProperty<IChannelBindingProvider>();
+                    if (cbp != null)
+                    {
+                        cbp.EnableChannelBindingSupport();
+                    }
+                }
+            }
+
+            void OnProtocolCreationComplete(SecurityProtocol securityProtocol)
+            {
+                SecurityProtocol = securityProtocol;
+                SecurityProtocol.ChannelParameters = _channelParameters;
+            }
+
+            public override T GetProperty<T>()
+            {
+                if (typeof(T) == typeof(ChannelParameterCollection))
+                {
+                    return (T)(object)_channelParameters;
+                }
+
+                return base.GetProperty<T>();
+            }
+        }
+
+        class SecurityOutputChannel : ClientSecurityChannel<IOutputChannel>, IOutputChannel
+        {
+            public SecurityOutputChannel(ChannelManagerBase factory, SecurityProtocolFactory securityProtocolFactory, IOutputChannel innerChannel, EndpointAddress to, Uri via)
+                : base(factory, securityProtocolFactory, innerChannel, to, via)
+            {
+            }
+
+            public IAsyncResult BeginSend(Message message, AsyncCallback callback, object state)
+            {
+                return BeginSend(message, DefaultSendTimeout, callback, state);
+            }
+
+            public IAsyncResult BeginSend(Message message, TimeSpan timeout, AsyncCallback callback, object state)
+            {
+                throw ExceptionHelper.PlatformNotSupported();   // Issue #31 in progress
+                //ThrowIfFaulted();
+                //ThrowIfDisposedOrNotOpen(message);
+                //return new OutputChannelSendAsyncResult(message, SecurityProtocol, InnerChannel, timeout, callback, state);
+            }
+
+            public void EndSend(IAsyncResult result)
+            {
+                throw ExceptionHelper.PlatformNotSupported();   // Issue #31 in progress
+                // OutputChannelSendAsyncResult.End(result);
+            }
+
+            public void Send(Message message)
+            {
+                Send(message, DefaultSendTimeout);
+            }
+
+            public void Send(Message message, TimeSpan timeout)
+            {
+                ThrowIfFaulted();
+                ThrowIfDisposedOrNotOpen(message);
+                TimeoutHelper timeoutHelper = new TimeoutHelper(timeout);
+                SecurityProtocol.SecureOutgoingMessage(ref message, timeoutHelper.RemainingTime());
+                InnerChannel.Send(message, timeoutHelper.RemainingTime());
+            }
+        }
```

Comment:
Yup, it's a blip on the radar.

---
In reply to: [78346337](https://github.com/dotnet/wcf/pull/1515#discussion_r78346337) [](ancestors = 78346337)